### DISCUSSION TITLE
Where should the polkadot provider live

### DISCUSSION THREAD

**Discussion by ryanleecode** [paritytech/polkadot-provider#3] [General]
@jsdw we have a separate repo for the spec. Should we keep it like this or move it to the substrate connect repo?

**Comment by jsdw** [comment]
My initial instinct here was to have everything relevant to building a conforming extension in one place, so that in the same repo we have:
- A template/example extension
- Definition of the interface one needs to implement to build a conforming extension
- Definition of the discovery protocol for finding extensions (since we are copying some eth spec we can just link to that though)
- A library that dapps can pull in to do this discovery and return conforming extensions.

However, I reflected a bit and I do also see the point in keeping this PolkadotProvider stuff in a separate repo at the moment, since this interface is, I think, also used for things other than extensions.

So maybe the right thing to do at the mo is to link to the relevant place in this repo when talking about the interface that extensions need to implement. (This is actually something I commented on the documentation issue in substrate-connect just now too).

I'd love to get input from Josep on this (weirdly can't link to the GitHub handle on my phone right now)!
